etl/steps/data/garden/neglected_tropical_diseases/2024-05-02/soil_transmitted_helminthiases.py

```diff
@@ -0,0 +1,137 @@
+        .drop_duplicates(subset=["country", "year"])
+    )
+    # Calculating the number of SAC treated
+    tb_nat_sac["estimated_number_of_sac_treated"] = (
+        tb["national_coverage__sac__pct"] * tb["population_requiring_pc_for_sth__sac"] / 100
+    )
+    tb_nat_pre_sac = (
+        tb[["country", "year", "national_coverage__pre_sac__pct", "population_requiring_pc_for_sth__pre_sac"]]
+        .copy()
+        .drop_duplicates()
+        .dropna(subset=["national_coverage__pre_sac__pct"])
+        .drop_duplicates(subset=["country", "year"])
+    )
+    # Calculating the number of pre-SAC treated
+    tb_nat_pre_sac["estimated_number_of_pre_sac_treated"] = (
+        tb["national_coverage__pre_sac__pct"] * tb["population_requiring_pc_for_sth__pre_sac"] / 100
+    )
+    # Adding region aggregates to selected variables
+    tb_nat_sac = add_regions_to_selected_vars(
+        tb_nat_sac,
+        cols=["country", "year", "population_requiring_pc_for_sth__sac", "estimated_number_of_sac_treated"],
+        ds_regions=ds_regions,
+    )
+    tb_nat_pre_sac = add_regions_to_selected_vars(
+        tb_nat_pre_sac,
+        cols=["country", "year", "population_requiring_pc_for_sth__pre_sac", "estimated_number_of_pre_sac_treated"],
+        ds_regions=ds_regions,
+    )
+    #  Splitting the table into two tables for pre-sac and sac
+    age_groups = ["pre_sac", "sac"]
+    tbs = {}
+    for age_group in age_groups:
+        cols = [
+            "country",
+            "year",
+            f"drug_combination__{age_group}",
+            f"number_of_{age_group}_targeted",
+            f"reported_number_of_{age_group}_treated",
+            f"programme_coverage__{age_group}__pct",
+        ]
+        tbs[f"tb_{age_group}"] = tb[cols].copy()
+        tbs[f"tb_{age_group}"].columns = [
+            "country",
+            "year",
+            "drug_combination",
+            "number_targeted",
+            "reported_number_treated",
+            "programme_coverage__pct",
+        ]
+        tbs[f"tb_{age_group}"] = tbs[f"tb_{age_group}"].dropna(
+            subset=[
+                "drug_combination",
+                "number_targeted",
+                "reported_number_treated",
+                "programme_coverage__pct",
+            ],
+        )
+        # There are some rows which seem to be erroneous duplicates, we will drop these e.g. Burundi 2015 for sac
+        tbs[f"tb_{age_group}"] = tbs[f"tb_{age_group}"].drop_duplicates(subset=["country", "year", "drug_combination"])
+
+    tb_pre_sac = tbs["tb_pre_sac"]
+    tb_sac = tbs["tb_sac"]
+    # Give the tables relevant short names
+    tb_pre_sac.metadata.short_name = "soil_transmitted_helminthiases_pre_sac"
+    tb_sac.metadata.short_name = "soil_transmitted_helminthiases_sac"
+    tb_nat_sac.metadata.short_name = "soil_transmitted_helminthiases_national_sac"
+    tb_nat_pre_sac.metadata.short_name = "soil_transmitted_helminthiases_national_pre_sac"
```

Comment:
Instead of assigning `short_name` directly, you can pass it as an argument of the function `Table.format`. E.g., for `tb_sac`:

```
tb_sac = tb_sac.format(["country", "year", "drug_combination"], short_name="soil_transmitted_helminthiases_sac")

Reply:
Thank you! I always forget this!